src/docfx/localization/LocalizationConvention.cs

```diff
@@ -146,8 +150,59 @@ public static bool TryResolveDocset(this Docset docset, string file, out Docset
                 return true;
             }
 
+            // try to resolve from source repo's git history
+            // todo: support code snippet
+            var fallbackDocset = GetFallbackDocset();
+            if (fallbackDocset != null && Document.GetContentType(file) == ContentType.Page)
+            {
+                var (gitCommitProvider, repoPath) = s_gitCommitProvider.GetOrAdd(fallbackDocset.DocsetPath, docsetPath =>
+                {
+                    var repo = GitUtility.FindRepo(docsetPath);
+                    if (!string.IsNullOrEmpty(repo))
+                    {
+                        var (remote, _, _) = GitUtility.GetRepoInfo(repo);
+                        return (GitCommitProvider.Create(repo), repo);
+                    }
+
+                    return default;
+                });
+
+                if (gitCommitProvider != null)
+                {
+                    var pathToRepo = PathUtility.NormalizeFile(Path.GetRelativePath(repoPath, Path.Combine(fallbackDocset.DocsetPath, file)));
+                    var commits = gitCommitProvider.GetCommitHistory(pathToRepo);
+                    if (commits.Count > 1)
+                    {
+                        // the latest commit would be deleting it from repo
+                        if (GitUtility.TryGetContentFromHistory(repoPath, pathToRepo, commits[1].Sha, out content))
```

Comment:
is this `commits[0]`?

Reply:
no, it's [1] since [0] is deleting the file

---
In reply to: [236457954](https://github.com/dotnet/docfx/pull/3677#discussion_r236457954) [](ancestors = 236457954)